lib/cocoapods/target/build_settings.rb

```diff
@@ -585,13 +588,20 @@ def initialize(target, non_library_spec = nil, configuration: nil)
         define_build_settings_method :static_frameworks_to_import, :memoized => true do
           static_frameworks_to_import = []
           static_frameworks_to_import.concat vendored_static_frameworks.map { |f| File.basename(f, '.framework') } unless target.should_build? && target.build_as_dynamic?
+          static_frameworks_to_import.concat vendored_xcframeworks
+                                                .select { |f| f.slices.any? { |slice| !Xcode::LinkageAnalyzer.dynamic_binary?(slice.binary_path) }}
+                                                .map(&:name)
+                                                .uniq unless target.should_build? && target.build_as_dynamic?
           static_frameworks_to_import << target.product_basename if target.should_build? && target.build_as_static_framework?
           static_frameworks_to_import
         end
 
         # @return [Array<String>]
         define_build_settings_method :dynamic_frameworks_to_import, :memoized => true do
           dynamic_frameworks_to_import = vendored_dynamic_frameworks.map { |f| File.basename(f, '.framework') }
+          dynamic_frameworks_to_import.concat vendored_xcframeworks
+                                                .select { |f| f.slices.any? { |slice| Xcode::LinkageAnalyzer.dynamic_binary?(slice.binary_path) }}
```

Comment:
and do we care if this is `any?` or `all?`?

Reply:
this is definitely a gray area - it would be strange to mix static and dynamic frameworks into a single .xcframework, but it _is_ possible. so we cannot assume that all slices have the same linkage